pkgs/by-name/un/undbx/package.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "undbx";
+  version = "0.22-unstable-2019-02-11";
+
+  src = fetchFromGitHub {
+    owner = "ZungBang";
+    repo = "undbx";
+    rev = "5e31c757e137a6409115cac0623d61d384019b7a";
+    hash = "sha256-leregcv3dv/D3WvFkYyjQePdKi4BgE0aj5PY6JiSKl8=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  makeFlags = [
+    "CC=cc"
+  ];
```

Comment:
```suggestion
```
I don't see a reason for this

Reply:
Darwin compatibility

Reply:
What happens without it? To me this looks like something `autoreconfHook` should handle though, if it is indeed required

Reply:
I will test it what happened on Darwin ofborg check

Reply:
> Some builds hardcode gcc/g++ in their build scripts, that can usually be fixed with using something like makeFlags = [ "CC=cc" ]; or by patching the build scripts.

Looking at the build files, I don't see hardcoded gcc

Reply:
as [Lurkki14](https://github.com/Lurkki14) said, this is unnecessary. the darwin build is failing for different reasons.
 > dbxread.c:178:9: error: variable 'n' set but not used [-Werror,-Wunused-but-set-variable]
    int n;
        ^
dbxread.c:571:31: error: operator '?:' has lower precedence than '+'; '+' will be evaluated first [-Werror,-Wparentheses]
        for (k1 = i + (i < k2)? 1:0;  k0 < k2;  k1 = (k0 + k2) / 2) {
                  ~~~~~~~~~~~~^

this is due to differences in warings between clang and gcc. can reproduce on linux by:
```console
nix-build -A pkgsLLVM.undbx
```

looking at the code it is unclear if `k1 = i + (i < k2)? 1:0` results in a bug or not.

Reply:
Ok. I will use only linux platform then.

Reply:
the error is not platform related, it is the compiler used and the build using `-Werror` which makes warnings into errors. one fix is to remove `-Werror` flag. usually, i like to report the issue upstream as well.

Reply:
You mean if we remove `-Werror` it will compile on Darwin correctly?

Reply:
you can build using clang on linux via `nix-build -A pkgsLLVM.undbx` to test.

Reply:
As agreed with the upstream developer, I applied the patch for it. What I need to understand is to check if that undbx.hta generated on bin directory is needed or not. It seems to be a Visual Basic file, maybe intended for Windows environment.

Reply:
The upstream developer confirmed me it is a file intended only for Windows. He suggested me just to comment the line
```
bin_SCRIPTS = undbx.hta
```
in Makefile.am.